### PR TITLE
Feature/choose beanstalk version

### PR DESCRIPTION
**Pull request by timdev**
Allow picking version via configuration.

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-  command: tar -zxvf /tmp/beanstalkd.tar.gz chdir=/tmp creates=/tmp/beanstalkd-1.9
+  command: tar -zxvf /tmp/beanstalkd.tar.gz chdir=/tmp creates=/tmp/beanstalkd-{{ beanstalkd_version }}
 
 - name: Make
-  command: make chdir=/tmp/beanstalkd-1.9/ creates=/tmp/beanstalkd-1.9/beanstalkd
+  command: make chdir=/tmp/beanstalkd-{{ beanstalkd_version}}/ creates=/tmp/beanstalkd-{{ beanstalkd_version}}/beanstalkd
 
 - name: Install
-  command: make install chdir=/tmp/beanstalkd-1.9/ creates=/usr/local/bin/beanstalkd
+  command: make install chdir=/tmp/beanstalkd-{{ beanstalkd_version}}/ creates=/usr/local/bin/beanstalkd
```
Comment (ramondelafuente) [inline tasks/main.yml:32]: Due to the `creates` line, the `make install` command will not run if you've already got 1.9
Not really a big issue, but having the variable creates the impression that this role will upgrade... 

Any thoughts?

Comment (timdev) [inline tasks/main.yml:32]: Good point.  Simply removing the `creates` seems ugly.   I'll give it some thought later and see if I can come up with a reasonable fix.

**Comment by timdev** [comment]
Having given this a bit more thought, I think removing the `creates` (and changing the final `state=started` to `state=restarted`) is probably the best option after all.

On the plus side, it would be mostly idempotent.  And it's simple.

On the downside, it's imperfect.  The daemon would be restarted even if no upgrade actually happened.  Considering the default of no persistence, that could be bad.

The other option would be using something like the [script module](http://docs.ansible.com/script_module.html) to upload and execute some script that would test for the presence of beanstalkd, check it's version, and then copy and (re)start as needed.

Any preference, or other ideas?

File: tasks/main.yml [hunk 2]
```diff
@@ -1,15 +1,15 @@
 ---
 - name: Get tarfile
-  command: wget https://github.com/kr/beanstalkd/archive/v1.9.tar.gz -q -O /tmp/beanstalkd.tar.gz creates=/tmp/beanstalkd.tar.gz
+  command: wget https://github.com/kr/beanstalkd/archive/v{{ beanstalkd_version }}.tar.gz -q -O /tmp/beanstalkd.tar.gz creates=/tmp/beanstalkd.tar.gz
```
Comment (ramondelafuente) [inline tasks/main.yml:7]: Also add the version to the tmp-file, otherwise newer versions won't get downloaded

**Comment by ramondelafuente** [comment]
I've been playing around with this, and it turned out the `Make install` command also did not honor the upgrading to a newer version. I've put together a working upgrade (and downgrade) version of your patch here:

https://github.com/f500/ansible-beanstalkd/commits/timdev-feature/choose-beanstalk-version

Could you take a look?

**Comment by timdev** [comment]
That ended up being a bit more work than I thought, mainly because I decided to allow configuration of the account the daemon runs under.  I hope you don't mind the scope creep, @ramondelafuente 

This PR now makes persistence the default, and allows setting or changing the user the daemon runs as.

Solving that made me realize that changing the storage location should also be handled more gracefully.  Currently, if you want to move the storage location, you'll lose the binlog.  It would be better to detect that change somehow, and try to move the storage directory instead of just creating it.

But that was the case before this PR anyway.  I think this can probably be merged as-is.  Let me know if you need anything else.

**Comment by ramondelafuente** [comment]
We've got a couple more roles that _try_ to handle such changes, but it's a headache because you need the old location as well as the new. We mostly just use beanstalkd as-is anyways, so let's wait until we need it or people actually start complaining ;-)

File: tasks/main.yml [hunk 3]
```diff
@@ -1,24 +1,49 @@
...
+- name: Ensure persistence storage location exists and has proper ownership.
+  file: path="{{ beanstalkd_storage }}" state=directory owner={{ beanstalkd_user }} group={{ beanstalkd_user }} recurse=true
+  register: storage_dir_changed
+  notify: restart beanstalkd
 
-- name: Install
-  command: make install chdir=/tmp/beanstalkd-1.9/ creates=/usr/local/bin/beanstalkd
+- name: Get tarfile.
+  command: wget https://github.com/kr/beanstalkd/archive/v{{ beanstalkd_version }}.tar.gz -q -O /tmp/beanstalkd-{{ beanstalkd_version }}.tar.gz creates=/tmp/beanstalkd-{{ beanstalkd_version }}.tar.gz
+  when: installed_beanstalkd_version.stdout != beanstalkd_version
```
Comment (ramondelafuente) [inline tasks/main.yml]: Did I add this 'when' ? I'd think the 'creates' handles this situation, and the when is overkill. On the other hand, the when skips the task, which should be quicker than the file-existance check...

File: tasks/main.yml [hunk 4]
```diff
@@ -1,24 +1,49 @@
...
 
-- name: Copy beanstalkd startup script
-  copy: src=beanstalkd.initd dest=/etc/init.d/beanstalkd mode=0755
+- name: Unarchive.
+  command: tar -zxvf /tmp/beanstalkd-{{ beanstalkd_version }}.tar.gz chdir=/tmp creates=/tmp/beanstalkd-{{ beanstalkd_version }}
+  when: installed_beanstalkd_version.stdout != beanstalkd_version
```
Comment (ramondelafuente) [inline tasks/main.yml]: Did I add this 'when' ? I'd think the 'creates' handles this situation, and the when is overkill. On the other hand, the when skips the task, which should be quicker than the file-existance check... (repeated)

File: tasks/main.yml [hunk 5]
```diff
@@ -1,24 +1,49 @@
...
 
-- name: Copy config
-  template: src=beanstalkd dest=/etc/default/beanstalkd
+- name: Make.
+  command: make chdir=/tmp/beanstalkd-{{ beanstalkd_version}}/ creates=/tmp/beanstalkd-{{ beanstalkd_version}}/beanstalkd
+  when: installed_beanstalkd_version.stdout != beanstalkd_version
```
Comment (ramondelafuente) [inline tasks/main.yml:27]: Did I add this 'when' ? I'd think the 'creates' handles this situation, and the when is overkill. On the other hand, the when skips the task, which should be quicker than the file-existance check... (repeated)

**Comment by ramondelafuente** [comment]
Oh, and I don't mind the scope-creep - thank _you_ for all the work on this!

File: tasks/main.yml (hunk 5 above)
Comment (timdev) [inline tasks/main.yml:27]: I added those 'when's while testing the owner-juggling stuff.  No reason to not skip them if the version isn't changing.

**Comment by timdev** [comment]
I agree it's not worth worrying about the storage location changing.  Probably best solved with a note in the documentation.

**Pull request merged**